scanpy/get.py

```diff
@@ -167,12 +170,36 @@ def obs_df(
         )
 
     # Make df
-    df = pd.DataFrame(index=adata.obs_names)
-    for k, l in zip(keys, lookup_keys):
-        if not use_raw or k in adata.obs.columns:
-            df[k] = adata.obs_vector(l, layer=layer)
+    df_var = df_obs = None
```

Comment:
I feel like this is a bit weird, I was not expecting `pd.concat` to allow some values to be `None`.

What about initializing the dataframe, then adding columns? So something like:

```python
df = pd.DataFrame(index=adata.obs_names)
...
# then adding columns like:
df[var_names] = matrix
...
df[obs_names] = adata.obs[obs_names]
...
# then reordering
df = df[keys]
```

Reply:
I just tried this second method but is significantly slower:

First time: current, second time: above solution. 

![image](https://user-images.githubusercontent.com/4964309/99650015-308bff00-2a55-11eb-84bc-92fae5d30ef4.png)

Reply:
That seems really strange.

Could you share the code you ran? I'd like to take a look and maybe open an issue in pandas.

Reply:
using `join` works equally fast. I will keep this solution instead. 

```PYTHON
df = pd.DataFrame(index=adata.obs.index)
... 
df = df.join(pd.DataFrame(matrix, columns=var_symbol, index=adata.obs.index))
```

`2.42 ms ± 195 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)`

Reply:
Never mind, I've got a small example now.

```python
In [22]: %timeit newdf = df.copy(); newdf[df2.columns] = df2
2.37 ms ± 31.2 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [23]: %timeit pd.concat([df, df2], axis=1)
215 µs ± 8.33 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

Reply:
https://github.com/pandas-dev/pandas/issues/37954